staging/vhost-device-gpu/src/virt_gpu.rs

```diff
@@ -458,9 +446,36 @@ impl VirtioGpu for RutabagaVirtioGpu {
         gpu_backend: &mut GpuBackend,
         gpu_scanout: VhostUserGpuScanout,
         resource_id: u32,
-        scanout_data: Option<VirtioScanoutBlobData>,
     ) -> VirtioGpuResult {
-        self.update_scanout_resource(gpu_backend, gpu_scanout, resource_id, scanout_data)
+        gpu_backend.set_scanout(&gpu_scanout).map_err(|e| {
+            error!("Failed to set scanout from frontend: {}", e);
+            ErrUnspec
+        })?;
+
+        // Virtio spec: "The driver can use resource_id = 0 to disable a scanout."
+        if resource_id == 0 {
+            error!("NOT IMPLEMENTED: disable scanout");
+            return Ok(OkNoData);
+        }
+
+        let _resource = self
+            .resources
+            .get_mut(&resource_id)
+            .ok_or(ErrInvalidResourceId)?;
+
+        //todo:
+        //create a display surface
+
+        // `resource_id` has already been verified to be non-zero
+        let resource_id = match NonZeroU32::new(resource_id) {
+            Some(id) => id,
+            None => return Ok(OkNoData),
+        };
+        //todo:
+        //store resource_id in a struct that will be used later
+        debug!("resource id: {:?}", resource_id);
+
+        Ok(OkNoData)
     }
```

Comment:
was there any need for this change in this commit? if not, I think the changes should follow the commit description.

Reply:
The commit description says the code is moved. But I guess the title of the message is confusing.

Reply:
ok, I see it in the description, it was a bit confusing due to the title. it's ok.